openoa/utils/downloader.py

```diff
@@ -0,0 +1,428 @@
+                        "area": [
+                            lat+node_spacing, lon-node_spacing, 
+                            lat-node_spacing, lon+node_spacing,
+                        ],
+                    },
+                    outfile)
+            except:
+                logger.error(f"Failed to download ERA5: {outfile}")
+
+
+    # get the saved data
+    ds_nc = xr.open_mfdataset(f"{save_pathname / f'{save_filename}*.nc'}")
+
+    # rename variables to conform with OpenOA
+    ds_nc = ds_nc.rename_vars({"si10":"windspeed_ms","t2m":"temperature_K","sp":"surf_pres_Pa"})
+    
+    # select the central node only for now
+    if 'expver' in ds_nc.dims:
+        sel = ds_nc.sel(expver=1,latitude=lat,longitude=lon, method="nearest")
+    else:
+        sel = ds_nc.sel(latitude=lat,longitude=lon, method="nearest")   
+
+    # convert to a pandas dataframe
+    df = sel.to_dataframe()
+
+    # select required columns
+    df = df[["windspeed_ms","temperature_K","surf_pres_Pa"]]
+
+    # rename the index to match other datasets
+    df.index.name = "datetime"
+
+    # drop any empty rows
+    df = df.dropna()
+
+    if save_filename is not None:
+        df.to_csv(save_pathname / f"{save_filename}.csv", index=True)
+
+    return df
+
+
+def get_merra2(
+        lat: float,
+        lon: float,
+        save_pathname: str | Path,
+        save_filename: str,
+        ) -> pd.DataFrame:
+    """
+    Get MERRA2 data directly from the NASA GES DISC service
+    This requires registration on the GES DISC service
+    See: https://disc.gsfc.nasa.gov/data-access#python-requests
+
+    Monthly 10m height data is demonstrated here,
+    as hourly data takes too long to download, but could be amended
+    and other GES DISC datasets also used (e.g. FLDAS)
+
+    Args:
+        lat(:obj:`float`): Latitude in WGS 84 spatial reference system (decimal degrees)
+        lon(:obj:`float`): Longitude in WGS 84 spatial reference system (decimal degrees)
+        save_pathname(:obj:`str` | :obj:`Path`): The path where the downloaded reanalysis data will
+            be saved
+        save_filename(:obj:`str`): The file name used to save the downloaded reanalysis data
+
+    Returns:
+        df(:obj:`dataframe`): A dataframe containing time series of the requested reanalysis variables
+        Saved NetCDF monthly MERRA2 files
+        Saved MERRA2 csv file
+    """
+    
+    logger.info("Please note access to MERRA2 data requires registration")
+    logger.info("Please see: https://disc.gsfc.nasa.gov/data-access#python-requests")
+
+    # base url containing the monthly data set M2IMNXLFO
+    base_url = r"https://goldsmr4.gesdisc.eosdis.nasa.gov/opendap/MERRA2_MONTHLY/M2IMNXLFO.5.12.4/"
+
+    # create save_pathname if it does not exist
+    save_pathname = Path(save_pathname).resolve()
+    if not save_pathname.exists():
+        save_pathname.mkdir()
+
+
+    now = datetime.datetime.now()
+    years = list(range(2000,now.year+1,1))
```

Comment:
Same comment about making `start_year` and `end_year` arguments to the function

Reply:
Done